src/libs/antares/study/parts/hydro/container.h

```diff
@@ -34,6 +34,104 @@
 
 namespace Antares::Data
 {
+/*!
+ ** \brief Final Reservoir Level data for a single area
+ */
+class FinalReservoirLevelRuntimeData
+{
+public:
+    // Final reservoir level runtime data
+
+    // vectors containing data necessary for final reservoir level calculation 
+    // for one area and all MC years
+    // vector indexes correspond to the MC years 
+    std::vector<bool> includeFinalReservoirLevel;
+    std::vector<double> endLevel;
+    std::vector<double> deltaLevel;
+
+    // Simulation Data
+    uint simEndDay;
+
+    // simulation year
+    // overwritten after each MC year
+    uint yearIndex;
+
+    // data per area
+    // data overwritten after each MC year 
+    double initialReservoirLevel;
+    double finalReservoirLevel;
+    double deltaReservoirLevel;
+    double reservoirCapacity;
+    double lowLevelLastDay;
+    double highLevelLastDay;
+    int initReservoirLvlMonth;
+
+    // methods:
+
+    void fillEmpty()
```

Comment:
Once all code from outside is here (from **hydro-final-reservoir-level-functions.cpp** I mean), create as many private methods as possible, because we don't want to expose too much details.
Indeed, as this class is only used in **management.cpp**, we could (roughly) have something like  (to be discussed) : 
```C++
void HydroManagement::prepareFinalReservoirLevelData(uint numSpace, uint year)
{
    study.areas.each([this, &numSpace, &year](Data::Area& area)
    {
          area.hydro.finalLevelInflowsModifyer.makeChecks()

          if (not area.hydro.finalLevelInflowsModifyer.isActive())
              return;

          area.finalLevelInflowsModifyer.updateInflows(year);
    });
}
```
What do you think ?

Reply:
Done very similarly to the proposed one. 
The data is first calculated and stored and then checked. So that is reversed.
return will break the area-loop so not used.